sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -332,8 +332,45 @@ case class FilterEstimation(plan: Filter) extends Logging {
         colStatsMap.update(attr, newStats)
       }
 
-      Some(1.0 / BigDecimal(ndv))
-    } else {
+      // We compute filter selectivity using Histogram information
+      attr.dataType match {
```

Comment:
use ` if (colStat.histogram.isEmpty)` to seperate the logic of basic stats (`Some(1.0 / BigDecimal(ndv))`) and histogram computation.